mumd/src/audio.rs

```diff
@@ -216,10 +217,43 @@ impl Audio {
 
         output_stream.play().unwrap();
 
-        let sounds = EVENT_SOUNDS
+        let mut res = Self {
+            output_config,
+            _output_stream: output_stream,
+            _input_stream: input_stream,
+            input_volume_sender,
+            input_channel_receiver: Some(input_receiver),
+            client_streams,
+            sounds: HashMap::new(),
+            output_volume_sender,
+            user_volumes,
+            play_sounds,
+        };
+        res.load_sound_effects(&[]);
+        res
+    }
+
+    pub fn load_sound_effects(&mut self, sound_effects: &[SoundEffect]) {
+        let overrides: HashMap<_, _> = sound_effects
             .iter()
-            .map(|(bytes, event)| {
-                let reader = hound::WavReader::new(*bytes).unwrap();
+            .filter_map(|sound_effect| {
+                let (event, file) = (&sound_effect.event, &sound_effect.file);
+                if let Ok(event) = NotificationEvents::try_from(event.as_str()) {
+                    Some((event, file))
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        self.sounds = NotificationEvents::iter()
+            .map(|event| {
+
+                let bytes = match overrides.get(&event) {
+                    Some(file) => get_sfx(file),
+                    None => get_default_sfx(),
+                };
```

Comment:
This might be cleaner to write as a combination of `Option::map` and `Option::unwrap_or_else`.